src/pages/api/datasets.ts

```diff
@@ -1,10 +1,33 @@
 import type {NextApiRequest, NextApiResponse} from 'next';
 import datasetFetcher from '@/lib/dataset-fetcher-instance';
+import {SearchOptions} from '@/lib/dataset-fetcher';
+
+interface ExtendedNextApiRequest extends NextApiRequest {
```

Comment:
I'm in doubt about the name `ExtendedNextApiRequest`. Does a user of the interface (even if it's just us) have to know that this is an extension of Next's API request? If the user is interested, he or she could inspect the definition and see that it `extends` another interface. Perhaps use a more 'functional' name, e.g. `DatasetApiRequest`?